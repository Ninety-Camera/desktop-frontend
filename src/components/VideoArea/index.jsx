import React from "react";
import { Box } from "@mui/system";
import { Stack } from "@mui/material";
import Grid from "@mui/material/Grid";
import VideoItem from "../VideoItem";

export default function VideoArea(props) {
  const videosList = props.videosList;
  const width = props.width;
  const height = props.height;
  const alignment = props.alignment;
  return (
    <Grid container spacing={5} justifyContent="center">
      {videosList.map((videoClip) => {
        return (
          <Grid item xs="auto" key={videosList.indexOf(videoClip)}>
            <VideoItem
              videoClip={videoClip}
              index={videosList.indexOf(videoClip)}
            />
          </Grid>
        );
      })}
    </Grid>
  );
}
