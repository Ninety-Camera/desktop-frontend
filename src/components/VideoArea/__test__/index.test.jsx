import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import ReactDom from "react-dom";
import VideoArea from "..";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { useState } from "react";
import VIDEOCLIP1 from "../../../assets/video1.mp4";

const videoList = [
  { sourcePath: VIDEOCLIP1, date: "20/02/2022", hour: "12:00" },
  { sourcePath: VIDEOCLIP1, date: "20/02/2022", hour: "12:00" },
  { sourcePath: VIDEOCLIP1, date: "20/02/2022", hour: "12:00" },
  { sourcePath: VIDEOCLIP1, date: "20/02/2022", hour: "12:00" },
  { sourcePath: VIDEOCLIP1, date: "20/02/2022", hour: "12:00" },
  { sourcePath: VIDEOCLIP1, date: "20/02/2022", hour: "12:00" },
];

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(<VideoArea videosList={videoList} alignment={"row"} />, div);
});

// it("renders form correctly", () => {
//   //use for props passing
//   const { getByTestId } = render(
//     <SnackBarComponent message="Successfully log in" />
//   );
//   expect(getByTestId("snackBarComponent")).toHaveTextContent(
//     "Successfully log in"
//   );
// });

it("matches snapshot", () => {
  const tree = renderer
    .create(<VideoArea videosList={videoList} alignment={"row"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
