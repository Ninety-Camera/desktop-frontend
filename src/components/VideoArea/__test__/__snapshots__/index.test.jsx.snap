// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1qgewj0-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
  >
    <img
      alt="CCTV video"
      height={240}
      onError={[Function]}
      src="video1.mp4"
      width={320}
    />
    <div
      className="css-aj204s-MuiStack-root"
    >
      <p
        style={
          Object {
            "color": "#6C63FF",
            "fontSize": 20,
            "fontWeight": 800,
            "margin": 0,
            "textAlign": "center",
          }
        }
      >
        CCTV - 1
      </p>
      <div
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          data-testid="toggleBtn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "textTransform": "none",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="StopIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6h12v12H6z"
              />
            </svg>
          </span>
          Stop Monitoring
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
  >
    <img
      alt="CCTV video"
      height={240}
      onError={[Function]}
      src="video1.mp4"
      width={320}
    />
    <div
      className="css-aj204s-MuiStack-root"
    >
      <p
        style={
          Object {
            "color": "#6C63FF",
            "fontSize": 20,
            "fontWeight": 800,
            "margin": 0,
            "textAlign": "center",
          }
        }
      >
        CCTV - 2
      </p>
      <div
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          data-testid="toggleBtn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "textTransform": "none",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="StopIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6h12v12H6z"
              />
            </svg>
          </span>
          Stop Monitoring
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
  >
    <img
      alt="CCTV video"
      height={240}
      onError={[Function]}
      src="video1.mp4"
      width={320}
    />
    <div
      className="css-aj204s-MuiStack-root"
    >
      <p
        style={
          Object {
            "color": "#6C63FF",
            "fontSize": 20,
            "fontWeight": 800,
            "margin": 0,
            "textAlign": "center",
          }
        }
      >
        CCTV - 3
      </p>
      <div
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          data-testid="toggleBtn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "textTransform": "none",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="StopIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6h12v12H6z"
              />
            </svg>
          </span>
          Stop Monitoring
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
  >
    <img
      alt="CCTV video"
      height={240}
      onError={[Function]}
      src="video1.mp4"
      width={320}
    />
    <div
      className="css-aj204s-MuiStack-root"
    >
      <p
        style={
          Object {
            "color": "#6C63FF",
            "fontSize": 20,
            "fontWeight": 800,
            "margin": 0,
            "textAlign": "center",
          }
        }
      >
        CCTV - 4
      </p>
      <div
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          data-testid="toggleBtn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "textTransform": "none",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="StopIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6h12v12H6z"
              />
            </svg>
          </span>
          Stop Monitoring
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
  >
    <img
      alt="CCTV video"
      height={240}
      onError={[Function]}
      src="video1.mp4"
      width={320}
    />
    <div
      className="css-aj204s-MuiStack-root"
    >
      <p
        style={
          Object {
            "color": "#6C63FF",
            "fontSize": 20,
            "fontWeight": 800,
            "margin": 0,
            "textAlign": "center",
          }
        }
      >
        CCTV - 5
      </p>
      <div
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          data-testid="toggleBtn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "textTransform": "none",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="StopIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6h12v12H6z"
              />
            </svg>
          </span>
          Stop Monitoring
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
  >
    <img
      alt="CCTV video"
      height={240}
      onError={[Function]}
      src="video1.mp4"
      width={320}
    />
    <div
      className="css-aj204s-MuiStack-root"
    >
      <p
        style={
          Object {
            "color": "#6C63FF",
            "fontSize": 20,
            "fontWeight": 800,
            "margin": 0,
            "textAlign": "center",
          }
        }
      >
        CCTV - 6
      </p>
      <div
        style={
          Object {
            "justifyContent": "center",
            "marginLeft": "auto",
            "marginRight": "auto",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium css-zcbmsk-MuiButtonBase-root-MuiButton-root"
          data-testid="toggleBtn"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "textTransform": "none",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="StopIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M6 6h12v12H6z"
              />
            </svg>
          </span>
          Stop Monitoring
        </button>
      </div>
    </div>
  </div>
</div>
`;
